### DISCUSSION TITLE
Is there any way to pass parameter to loader function

### DISCUSSION THREAD

**Discussion by f-z-coder** [remix-run/remix#6819] [Q&A]
### async function loadScript(apiKey) {
  const loader = new Loader({
    apiKey: apiKey,
    version: "weekly",
    region: "IN",
    libraries: ["places"],
  });
  return loader;
}

this is my loader function which require parameter   

const router = createBrowserRouter([
  {
    path: "/",
    element: <App />,
    loader:loadScript,
  },
]);

**here in router i want to pass my apikey with my loader function   ,is there any way to do that**

**Comment by lili21** [comment]
is it react-router? if you are using remix, you can use env variables.

**Reply by f-z-coder** [reply]
I am using react-router-dom v6.15   

I want to pass parameters to loader function(loadScript).   

I there any way to pass argument to loader function

**Reply by lili21** [reply]
maybe you should open the discussion in the react-router repo.

**Reply by lili21** [reply]
```js
const router = createBrowserRouter([
  {
    path: "/",
    element: ,
    loader:loadScript(apiKey),
  },
]);

function loadScript(apiKey) {

  return ({ request, params }) => {
    // you can access the apiKey inside this function
  }
}
```

**Reply by lili21** [reply]
But I would recommend you store all the arguments in the url as long as you can.

**Reply by f-z-coder** [reply]
I think we have to pass loader function not call loader function loadScript with(api key ) call the  is it write?

**Reply by lili21** [reply]
loadScript is not loader function, it is a high order function the return the loader function. so you can call it.

**Reply by sergiodxa** [reply]
Is this API key shared between users? Or unique per user?

If it's unique per user, store it somewhere else (like a cookie or localStorage), then in the loader read from there.

IF it's shared between users, either hard code it in the code, or send it in the initial HTML, or get it (e.g. with a fetch) before you even render the app, once you have it, store it globally and get it in the loader from that global place (e.g. localStorage).

**Comment by f-z-coder** [comment]
And have to pass argent that only access in component like map obj reference
